diff --git a/browser/components/preferences/preferences.js b/browser/components/preferences/preferences.js
index 7e21bab426..ebf3ac61b4 100644
--- a/browser/components/preferences/preferences.js
+++ b/browser/components/preferences/preferences.js
@@ -197,6 +197,7 @@ function init_all() {
   Preferences.queueUpdateOfAllElements();
   Services.telemetry.setEventRecordingEnabled("aboutpreferences", true);
 
+  register_module("paneCsk", {init(){}});
   register_module("paneGeneral", gMainPane);
   register_module("paneHome", gHomePane);
   register_module("paneSearch", gSearchPane);
