diff --git a/browser/components/sessionstore/TabState.sys.mjs b/browser/components/sessionstore/TabState.sys.mjs
index 26f5671c84..38bca452c7 100644
--- a/browser/components/sessionstore/TabState.sys.mjs
+++ b/browser/components/sessionstore/TabState.sys.mjs
@@ -87,6 +87,8 @@ var TabStateInternal = {
     let tabData = { entries: [], lastAccessed: tab.lastAccessed };
     let browser = tab.linkedBrowser;
 
+    tabData.floorpDisableHistory = tab.getAttribute("floorp-disablehistory");
+
     if (tab.pinned) {
       tabData.pinned = true;
     }
