diff --git a/remote/webdriver-bidi/WebDriverBiDiConnection.sys.mjs b/remote/webdriver-bidi/WebDriverBiDiConnection.sys.mjs
index ab1fc206f3..01933d5bc4 100644
--- a/remote/webdriver-bidi/WebDriverBiDiConnection.sys.mjs
+++ b/remote/webdriver-bidi/WebDriverBiDiConnection.sys.mjs
@@ -204,6 +204,11 @@ export class WebDriverBiDiConnection extends WebSocketConnection {
         result = await this.session.execute(module, command, params);
       }
 
+      if (module === "session" && command === "new") {
+        result = JSON.parse(JSON.stringify(result))
+        result.capabilities.browserName = "firefox";
+      }
+
       this.sendResult(id, result);
 
       // Session clean up.
