diff --git a/browser/moz.configure b/browser/moz.configure
index e8b401a7df..5ac00258a0 100644
--- a/browser/moz.configure
+++ b/browser/moz.configure
@@ -4,6 +4,8 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+include("../noraneko/static/gecko/config/moz.configure")
+
 imply_option("MOZ_PLACES", True)
 imply_option("MOZ_SERVICES_HEALTHREPORT", True)
 imply_option("MOZ_SERVICES_SYNC", True)
