diff --git a/toolkit/mozapps/installer/windows/nsis/common.nsh b/toolkit/mozapps/installer/windows/nsis/common.nsh
index a5e3977fb0..d270753bb4 100755
--- a/toolkit/mozapps/installer/windows/nsis/common.nsh
+++ b/toolkit/mozapps/installer/windows/nsis/common.nsh
@@ -538,7 +538,7 @@
 
       ; Add our subdirectory, this is hardcoded as grandparent of the update directory in
       ; several other places.
-      StrCpy $0 "$0\Mozilla-1de4eec8-1241-4177-a864-e594e8d1fb38"
+      StrCpy $0 "$0\Noraneko-1de4eec8-1241-4177-a864-e594e8d1fb38"
 
       Exch $0   ; Restore original $0 and put our $0 on the stack.
     FunctionEnd
