diff --git a/browser/components/extensions/parent/ext-sidebarAction.js b/browser/components/extensions/parent/ext-sidebarAction.js
index 2bdfe12f9ebde..1baec2ffbc4f3 100644
--- a/browser/components/extensions/parent/ext-sidebarAction.js
+++ b/browser/components/extensions/parent/ext-sidebarAction.js
@@ -49,6 +49,24 @@ this.sidebarAction = class extends ExtensionAPI {
     };
     this.globals = Object.create(this.defaults);
 
+    /****************************************** Floorp ************************************************************************************/
+    let prefObj = JSON.parse(Services.prefs.getStringPref("floorp.extensions.webextensions.sidebar-action"));
+
+    const keys = Object.keys(this.defaults.icon);
+    const lastKey = keys[keys.length - 1];
+    const lastURL = this.defaults.icon[lastKey];
+
+    if (!prefObj.data[extension.id]) {
+      prefObj.data[extension.id] = {
+        title: this.defaults.title,
+        panel: this.defaults.panel,
+        icon: lastURL,
+      }
+    }
+
+    Services.prefs.setStringPref("floorp.extensions.webextensions.sidebar-action", JSON.stringify(prefObj));
+    /**************************************************************************************************************************************/
+
     this.tabContext = new TabContext(target => {
       let window = target.ownerGlobal;
       if (target === window) {
