diff --git a/browser/components/tabbrowser/content/tabbrowser.js b/browser/components/tabbrowser/content/tabbrowser.js
index 14de79b543..8b2a27e594 100644
--- a/browser/components/tabbrowser/content/tabbrowser.js
+++ b/browser/components/tabbrowser/content/tabbrowser.js
@@ -3279,6 +3279,27 @@
         let tab;
         let tabWasReused = false;
 
+        /*@nora:inject:start*/
+        if (tabData.floorpDisableHistory) {
+          continue;
+        }
+
+        let floorpWorkspaceId,
+          floorpLastShowWorkspaceId,
+          floorpSSB;
+
+
+        floorpWorkspaceId = tabData.floorpWorkspaceId ? tabData.floorpWorkspaceId : JSON.parse(
+          Services.prefs.getStringPref("floorp.workspaces.v3.data")
+        ).workspaces[0].id
+        floorpLastShowWorkspaceId = tabData.floorpLastShowWorkspaceId;
+        floorpSSB = tabData.floorpSSB;
+
+        if (floorpSSB) {
+          window.close();
+        }
+        /*@nora:inject:end*/
+
         // Re-use existing selected tab if possible to avoid the overhead of
         // selecting a new tab. For now, we only do this for horizontal tabs;
         // we'll let tabs.js handle pinning for vertical tabs until we unify
@@ -3291,6 +3312,27 @@
         ) {
           tabWasReused = true;
           tab = this.selectedTab;
+
+          /*@nora:inject:start*/
+          if (floorpWorkspaceId) {
+            tab.setAttribute(
+              "floorpWorkspaceId",
+              floorpWorkspaceId
+            );
+          }
+
+          if (floorpLastShowWorkspaceId) {
+            tab.setAttribute(
+              "floorpWorkspaceLastShowId",
+              floorpLastShowWorkspaceId
+            );
+          }
+
+          if (floorpSSB) {
+            tab.setAttribute("floorpSSB", floorpSSB);
+          }
+          /*@nora:inject:end*/
+
           if (!tabData.pinned) {
             this.unpinTab(tab);
           } else {
@@ -3340,6 +3382,22 @@
             preferredRemoteType,
           });
 
+          /*@nora:inject:start*/
+          if (floorpWorkspaceId) {
+            tab.setAttribute(
+              "floorpWorkspaceId",
+              floorpWorkspaceId
+            );
+          }
+
+          if (floorpLastShowWorkspaceId) {
+            tab.setAttribute(
+              "floorpWorkspaceLastShowId",
+              floorpLastShowWorkspaceId
+            );
+          }
+          /*@nora:inject:end*/
+
           if (select) {
             tabToSelect = tab;
           }
@@ -4234,6 +4292,16 @@
         aTab,
         this
       );
+
+       /*@nora:inject:start*/
+      // Force to close & Make do not save history of the tab.
+      try {
+        this._endRemoveTab(aTab);
+      } catch (e) {
+        console.warn(e);
+      }
+      /*@nora:inject:end*/
+
     },
 
     _hasBeforeUnload(aTab) {
