diff --git a/browser/components/newtab/lib/TopSitesFeed.sys.mjs b/browser/components/newtab/lib/TopSitesFeed.sys.mjs
index a527a457678a7..45101c5dc45d7 100644
--- a/browser/components/newtab/lib/TopSitesFeed.sys.mjs
+++ b/browser/components/newtab/lib/TopSitesFeed.sys.mjs
@@ -699,6 +699,16 @@ export class TopSitesFeed {
     let hasContileTiles = false;
     if (contileEnabled) {
       let contilePositionIndex = 0;
+      let sponsorsList = [
+        {
+          id: 0,
+          name: "CubeSoft",
+          url: "https://www.cube-soft.jp",
+          image_url:
+            "https://raw.githubusercontent.com/cube-soft/cube.assets/master/cubesoft/logo/256px.png",
+          image_size: 200,
+        },
+      ];
       // We need to loop through potential spocs and set their positions.
       // If we run out of spocs or positions, we stop.
       // First, we need to know which array is shortest. This is our exit condition.
@@ -707,8 +717,8 @@ export class TopSitesFeed {
         this._contile.sites.length
       );
       // Loop until we run out of spocs or positions.
-      for (let i = 0; i < minLength; i++) {
-        let site = this._contile.sites[i];
+      for (let i = 0; i < sponsorsList.length; i++) {
+        let site = sponsorsList[i];
         let hostname = shortURL(site);
         let link = {
           isDefault: true,
