diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index b3ab5e8ca5..c0bc9929fc 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -189,6 +189,10 @@
 @RESPATH@/chrome/remote.manifest
 #endif

+; [Noraneko Files]
+@RESPATH@/chrome/noraneko.manifest
+@RESPATH@/chrome/noraneko@JAREXT@
+
 ; [Extensions]
 @RESPATH@/components/extensions-toolkit.manifest
 @RESPATH@/browser/components/extensions-browser.manifest
@@ -452,6 +456,3 @@ bin/libfreebl_64int_3.so
 ; Test workers unpack artifact build packages and need `jogfile.json`
 @RESPATH@/jogfile.json
 #endif
-
-; [Noraneko Files]
-@RESPATH@/noraneko/*
