diff --git a/remote/components/Marionette.sys.mjs b/remote/components/Marionette.sys.mjs
index 87a53679e4..8c1552d21d 100644
--- a/remote/components/Marionette.sys.mjs
+++ b/remote/components/Marionette.sys.mjs
@@ -140,7 +140,7 @@ class MarionetteParentProcess {
             Services.obs.addObserver(this, "domwindowopened");
           }
 
-          lazy.RecommendedPreferences.applyPreferences();
+          //lazy.RecommendedPreferences.applyPreferences();
 
           // Only set preferences to preserve in a new profile
           // when Marionette is enabled.
